### DISCUSSION TITLE
Is it possible to communicate with mysql2 from an html client open to all?

### DISCUSSION THREAD

**Discussion by Madriix** [sidorares/node-mysql2#1653] [Announcements]
Currently with my html client open to all (created with NodeJS) I use mysql2 but only in the URL GET or URL POST side, and I would like to know if it would be possible with an html client to communicate with mysql2 and this without passing by GET/POST by URL

The goal is to make the client cleaner, faster, more secure, without the inconvenience of get/post url

**Comment by sidorares** [comment]
can you elaborate what do you call "html client"? Is that a browser? Is so the simple answer is "no" - you can't open a raw tcp connection from html page opened in a browser. The closest you can get is to start some sort of a websocket proxy that translates tcp packets into websocket frames and vice verse

mysql2 code supposed to be run from node.js ( or node.js like ) runtime, for example you currently can't use it from a script started with bun.js due to `net` module not supported yet - see [node.js compatibility / "net" module](https://github.com/oven-sh/bun/issues/159)
